mne/viz/topo.py

```diff
@@ -250,15 +250,28 @@ def _imshow_tfr(ax, ch_idx, tmin, tmax, vmin, vmax, onselect, ylim=None,
                 tfr=None, freq=None, vline=None, x_label=None, y_label=None,
                 colorbar=False, picker=True, cmap=('RdBu_r', True), title=None,
                 hline=None):
-    """Show time-freq map on topo."""
+    """Show time-freq map as 2d image."""
     import matplotlib.pyplot as plt
     from matplotlib.widgets import RectangleSelector
 
-    extent = (tmin, tmax, freq[0], freq[-1])
     cmap, interactive_cmap = cmap
+    extent = (tmin, tmax, freq[0], freq[-1])
+    times = np.linspace(tmin, tmax, num=tfr[ch_idx].shape[1])
+
+    # construct a grid of borders between time/frequency bins
+    time_diff = np.diff(times) / 2.
+    freq_diff = np.diff(freq) / 2.
+    time_lims = np.concatenate([[times[0] - time_diff[0]], times[:-1] +
+                               time_diff, [times[-1] + time_diff[-1]]])
+    freq_lims = np.concatenate([[freq[0] - freq_diff[0]], freq[:-1] +
+                               freq_diff, [freq[-1] + freq_diff[-1]]])
```

Comment:
Nice.

...but this will break if there is only one frequency. Either raise an error in that case, or (what I would do) just plot from `freq - 0.5, freq + 05`. I think you can get this by just doing a conditional `freq_diff = [0.5] if len(freq) < 2 else np.diff(freq) / 2.` Same comment about `time_diff`.